model.py

```diff
@@ -1899,16 +1899,25 @@ def find_existing_identifiers(identifier_details):
         # Find identifiers that are already in the database.
         find_existing_identifiers(identifier_details.values())
 
+        # Remove the existing identifiers from the identifier_details list,
+        # regardless of whether the provided URN was accurate.
+        existing_details = [(i.type, i.identifier) for i in identifiers_by_urn.values()]
```

Comment:
You can eliminate the duplicate check further down by turning this into a `set`.